### DISCUSSION TITLE
Access files uploaded to filesystem after build

### DISCUSSION THREAD

**Discussion by hrafaelveloso** [vercel/next.js#14769] [Help] [answered]
Hi guys!

I've this problem that I can't figure a way of solving it...

So, I've one NextJS app built with a custom express server. One of my routes is responsible to handle image uploads to the server and then saving them in public folder.

During development I made some tests and I had some files in `public/images/articles folder`.

I ran `next build` and everything works well in `production`. If I upload some file in `production`, the file is being saved correctly, however, when I go to `https://<myurl>/images/articles/<new file added>` I get a `404`. The file is in the filesystem but it's not being served by my app.

The files that existed in public folder are served right. Is there anything that I should do in my server file to serve files from public that were added after the current build?

Thanks!

**Comment by jamesmosier** [comment, marked as answer]
If you alter your Express server to allow for uploaded files to be served via Express from the `public` folder that will likely break Next's existing functionality with the `public` folder. So I would say you should consider uploading/serving them from a different folder outside of Next.

If you want to upload & serve files from another directory you could use `app.use(express.static('files'));` (assuming you have a `files` folder) and then they would be accessible at `domain.com/filename.jpg`

[Express static docs](https://expressjs.com/en/starter/static-files.html)

**Reply by hrafaelveloso** [reply]
Thanks for your answer @jamesmosier ! 

I thought that I could use `public` folder, but now I moved the uploaded files to `files` folder!

**Comment by martinratinaud** [comment]
@hrafaelveloso can you please share a basic config of your custom server ? and how you use it in package.json please?

**Comment by hc0503** [comment]
How can I access to upload folder after building?
@hrafaelveloso , let me know if you already resolved that issue.

**Comment by muserref-sefer** [comment]
Actually i added files folder but still i cannot access to files of this folder. Where i can add this folder? @jamesmosier @hrafaelveloso 

```
--server.js
--public
--files
    --image.png
```

server.js
```
const next = require( 'next' );
const express = require( 'express');

const port = 3000;
const dev = process.env.NODE_ENV !== 'production';
const app = next( { dev } );

app.prepare()
  .then( () => {
    const server = express();

    server.use(express.static('files'));

    server.listen( port, ( err ) => {
      if ( err ) {
        throw err;
      }
      console.warn( `Ready on http://localhost:${port}` );
    });
  });
```
